modules/claims_api/lib/bd/bd.rb

```diff
@@ -100,12 +100,18 @@ def get_claim_id(doc_type, claim)
     #
     # @return {parameters, file}
     # rubocop:disable Metrics/ParameterLists
-    def generate_upload_body(claim:, doc_type:, pdf_path:, action:, file_number: nil, original_filename: nil,
+    def generate_upload_body(claim:, doc_type:, pdf_path:, action:, file_number: nil, original_filename: nil, # rubocop:disable Metrics/MethodLength
                              pctpnt_vet_id: nil)
       payload = {}
       auth_headers = claim.auth_headers
       veteran_name = compact_veteran_name(auth_headers['va_eauth_firstName'], auth_headers['va_eauth_lastName'])
-      birls_file_num = auth_headers['va_eauth_birlsfilenumber'] || file_number if doc_type != 'L705'
+      if doc_type == 'L122'
+        birls_file_num = auth_headers['va_eauth_birlsfilenumber']
+      elsif doc_type != 'L705'
+        birls_file_num = file_number
+      elsif %w[L075 L190].include?(doc_type)
+        birls_file_num = nil
+      end
```

Comment:
do you think it would be worth it to move this out into a method to handle setting the doc_type?  `generate_upload_body` is starting to feel kind of messy and cluttered.

Reply:
Good call, moved out to separate method.